storage/client.go

```diff
@@ -48,6 +49,10 @@ const (
 	userAgentHeader = "User-Agent"
 )
 
+var (
+	validStorageAccount = regexp.MustCompile("^[0-9a-z]{3,24}$")
```

Comment:
I don't want folks to have to duplicate this logic in their code, and I can imagine folks wanting to validate before making the NewClient call. Would something like this be too fancy? [Go Playground](https://play.golang.org/p/malkhcYxzr)

If it is too fancy, maybe just expose a method named `ValidAccountName(name string) bool`